### ISSUE TITLE
force does not clear out contents of current release

### ISSUE THREAD

**Issue by choubacha** [RiotGamesCookbooks/artifact-cookbook#28]
If the force option is specified, the current release directory is not rm -rfed. The implication is that if my tarball has files removed from it and I redeploy, I expect the files to be removed from the installed directory. Since the files are not removed, this expectation is not met.

My recommendation is to change:

``` ruby
action :deploy do
  next unless new_resource.force or not deployed?
```

to

``` ruby
action :deploy do
  delete_release_path_for(version.basename) if force
  next unless deployed?
```

There will probably need to be some other changes, but this is the simplest approach.

**Comment by KAllan357** [comment]
Tagging this for a 1.0.0 for now and I will think about it a bit.

**Comment by KAllan357** [comment]
I'm going to omit this from a 1.0.0 release. While this issue may be a pain in the ass during development / iteration, I don't think it is one you'd see in a production cookbook.

According to most descriptions of Continuous Delivery I've seen, each artifact you create should have a unique version identifier. If your tarball is changing between `artifact_deploy` executions, I believe that should be considered a new version of that artifact and deployed into a new directory.

**Comment by ivey** [comment]
I disagree about `rm -rf` and I disagree that it should happen on force. To handle this right for the 99% who aren't deploying different code with the same version _and_ to accommodate the 1% who are making a huge mistake, this should be a separate attribute, `:remove_on_force`, and it should be a `mv` in case you need to recover from it.

**Comment by KAllan357** [comment]
I'd like to come back to this for 1.6.0. I like the suggestions by @ivey, but I think we can do without recovery aspect.

**Comment by ivey** [comment]
Would be a great feature to TDD.

**Comment by KAllan357** [comment]
Adding commits to fix this to #64
